SPID Docs conformance

damikael opened the pull request:
## Title
Allineamento a Documentazione SPID emanata

## Content
Allineamenti e disambiguazioni proprie per SPID in accordo alla documentazione ufficiale emanata

## Review

- [X] Ensure your files are written following RST specs (*not MD!*)
- [X] Italian version
- [X] English version
- [ ] Example files 
- [X] Ask for review

damikael committed:
Update metadata_oidc_op.rst

upd description on
request_object_encryption_alg_values_supported
request_object_encryption_enc_values_supported

damikael committed:
Update metadata_oidc_op.rst

Aggiornata descrizione per
request_object_encryption_alg_values_supported
request_object_encryption_enc_values_supported

damikael committed:
Update metadata_oidc_rp.rst

add response_types that MUST contain only the value 'code'

damikael committed:
Update metadata_oidc_rp.rst

add response_types that MUST contain only the value 'code'.

damikael committed:
Update authorization_endpoint.rst

add ui_locales for SPID

damikael committed:
Update authorization_endpoint.rst

add ui_locales for SPID

damikael committed:
Update token_endpoint.rst

ID token is a signed JWT

damikael committed:
Update token_endpoint.rst

ID Token is a signed JWT

damikael committed:
Update token_endpoint.rst

fix typ header claim on access_token

damikael committed:
Update token_endpoint.rst

fix typ header claim on access_token

damikael committed:
Update token_endpoint.rst

fix: access_token must be pairwise type

damikael committed:
Update revocation_endpoint.rst

when the user session on the RP expires, the RP MUST revoke the Access Token in its possession

damikael committed:
Update revocation_endpoint.rst

Quando l'utente esegue il logout o quando la sua sessione presso il RP scade (in base alle policy decise da quest'ultimo) il RP DEVE chiamare questo endpoint per revocare l’access token e l’eventuale refresh token in suo possesso.

damikael committed:
Update revocation_endpoint.rst

damikael committed:
Update revocation_endpoint.rst

damikael committed:
Merge pull request #134 from italia/damikael-patch-8

Damikael patch 8

damikael committed:
Merge pull request #135 from italia/damikael-patch-7

Damikael patch 7

damikael committed:
Merge pull request #136 from italia/damikael-patch-6

Damikael patch 6

damikael committed:
Merge pull request #137 from italia/damikael-patch-5

Damikael patch 5

damikael committed:
Merge pull request #138 from italia/damikael-patch-4

Damikael patch 4

damikael committed:
Merge pull request #139 from italia/damikael-patch-3

Damikael patch 3

damikael committed:
Merge pull request #140 from italia/damikael-patch-2

Damikael patch 2

damikael committed:
Merge pull request #142 from italia/damikael-patch-1

Damikael patch 1

damikael committed:
Update metadata_oidc_rp.rst

fix description of response_types

damikael committed:
Update revocation_endpoint.rst

damikael committed:
Update metadata_oidc_rp.rst

peppelinux commented on docs/en/revocation_endpoint.rst:
```diff
@@ -5,12 +5,12 @@ Revocation Endpoint
   The Access Token revocation implies revoking all the Refresh Tokens linked to it.
 
-The OP will have to revoke the token specified in the request and will have to end the Single Sign-On session, if it is still active.
+The OP will have to revoke the token specified in the request and will have to end the Single Sign-On session, if it is still active. Any other active tokens for the user must be kept valid.
```
is Single Sign-On a defined term?

would we like to define SSO in the defined terms to give our assumption on its meaning?

fmarino-ipzs replied:
Questi contenuti andranno gestiti con box di disambiguazione sulla base anche della PR #131

AntonioFlorio replied:
Nel prossimo meeting dovremmo affrontare il problema della definizione di "sessione di SSO" e di "SSO".

fmarino-ipzs commented on docs/en/metadata_oidc_rp.rst:
```diff
@@ -95,6 +95,9 @@ The RP Metadata of type **"openid_relying_party"** MUST contain at least the fol
   * - **client_registration_types**
     - See `OIDC-FED`_ Section 4.1. The required value is **automatic**. 
     - |spid-icon| |cieid-icon|
+  * - **response_types**
```
Non era stato inserito perchè opzionale e il suo valore di default è proprio code.
response_types
OPTIONAL. JSON array containing a list of the OAuth 2.0 response_type values that the Client is declaring that it will restrict itself to using. If omitted, the default is that the Client will use only the code Response Type.

Inoltre nel metadata OP è presente il valore response_type_supported che DEVE essere valorizzato con code. 

Dal mio punto di vista aggiungerlo è superfluo

peppelinux replied:
effettivamente un RP può partecipare a federazioni multiple e malauguratamente supportare altri flussi oltre a code

@damikael leggo da LLGG SPID OIDC che "Deve" (non uppercased, quindi normativo-debole)

@fmarino-ipzs vogliamo specificare che DEVE solo per SPID qui oppure va bene DEVE anche per CIE?

fmarino-ipzs replied:
@peppelinux , se effettivamente un RP può partecipare ad altre federazioni che supportano altri flussi oltre a code, forse sarebbe meglio riportare "It MUST contain the value **code**." (senza "only"). Che ne pensi? 
Per quanto riguarda la CIE per me va bene anche il MUST, tanto, come già detto, quello che poi conta è il response_type_supported nel metadata OP.

peppelinux replied:
eccolo afca7d53462c76b30cce90390b01320f9220209f

fmarino-ipzs commented on docs/en/token_endpoint.rst:
```diff
@@ -226,7 +226,7 @@ UserInfo endpoint to get user attributes.
 ++++++++
 
-The ID Token is a JSON Web Token (JWT) that contains information on the user that has executed the authentication. The RPs MUST validate the ID Token.
+The ID Token is a signed JSON Web Token (JWT) that contains information on the user that has executed the authentication. The RPs MUST validate the ID Token.
```
on the user that has executed the authentication --> about the user that has performed the authentication

fmarino-ipzs commented on docs/it/metadata_oidc_rp.rst:
```diff
@@ -96,6 +96,9 @@ Il Metadata di tipo **"openid_relying_party"** DEVE contenere almeno i seguenti
   * - **client_registration_types**
     - Vedi `OIDC-FED`_ Section 4.1. Il valore richiesto è **automatic**. 
     - |spid-icon| |cieid-icon|
+  * - **response_types**
```
Non era stato inserito perchè opzionale e il suo valore di default è proprio code.
response_types
OPTIONAL. JSON array containing a list of the OAuth 2.0 response_type values that the Client is declaring that it will restrict itself to using. If omitted, the default is that the Client will use only the code Response Type.

Inoltre nel metadata OP è presente il valore response_type_supported che DEVE essere valorizzato con code.

Dal mio punto di vista aggiungerlo è superfluo

AntonioFlorio replied:
Lo abbiamo aggiunto per evitare che qualcuno lo possa inserire con un valore diverso da code.

peppelinux replied:
per me sarebbe SHOULD e non MUST

questo aspetto sta nei QaD tests AgID degli RP?

fmarino-ipzs commented on docs/it/revocation_endpoint.rst:
```diff
@@ -5,12 +5,12 @@ Revocation Endpoint
   La revoca di un Access Token comporta la revoca di tutti i Refresh Token a questo collegati.
 
-L'OP DOVRÀ revocare il token specificato nella richiesta e DOVRÀ terminare la sessione di Single Sign-On se ancora attiva.
+L'OP DOVRÀ revocare il token specificato nella richiesta e DOVRÀ terminare la sessione di Single Sign-On se ancora attiva. Eventuali altri token attivi per l’utente dovranno invece essere mantenuti validi.
```
Questi contenuti andranno gestiti con box di disambiguazione sulla base anche della PR https://github.com/italia/spid-cie-oidc-docs/pull/131

fmarino-ipzs reviewed (approved):
A parte una piccola nota di carattere editoriale sulla versione inglese, per me si può unire. I contenuti presenti nella sezione Revocation Endpoint andranno poi rivisti successivamente per disambiguare il comportamento di CIE e SPID così come proposto nella PR #131

damikael committed:
Update log_management.rst

add spid admonition

damikael committed:
Update log_management.rst

add spid admonition

peppelinux commented:
Cari questi commit anonimi li squashio

Nel testo della pr dovremmo avere un sommario, una brevissimo elenco puntato delle modifiche fatte, dato che mancano queste informazioni nei messaggi dei commit

L'uso di git rende il changelog delle modifiche trasparente, è per i nostri lettori ed utenti farlo, ma anche per noi

damikael committed:
upd link on long revocable session to guidelines

damikael committed:
Merge pull request #145 from italia/damikael-patch-1

Damikael patch 1

damikael committed:
upd link on long revocable session to guidelines page 47

damikael committed:
upd federation historical key history

the TA MUST publish **its** federation public key history

as for 

https://openid.net/specs/openid-connect-federation-1_0.html#name-federation-historical-keys-

damikael committed:
upd Registro storico

il TA DEVE pubblicare lo storico delle proprie chiavi pubbliche (JWKS)

come indicato in

https://openid.net/specs/openid-connect-federation-1_0.html#name-federation-historical-keys-

damikael committed:
upd Resolve endpoint

This endpoint MUST release the Metadata, the Trust Marks and the previously processed Trust Chain, and MUST
NOT trigger a procedure of Federation Entity Discovery for each request arrival** , unless this endpoint is secured with a client authentication mechanism, such as private_key_jwt [OIDC-CORE]**.

damikael committed:
upd Resolve endpoint

Questo endpoint DEVE rilasciare i Metadata, i Trust Mark e la Trust Chain già precedentemente elaborata e NON DEVE innescare una procedura di Federation Entity Discovery ad ogni richiesta pervenuta**, a meno che questo endpoint non venga protetto con un meccanismo di autenticazione dei client, come ad esempio private_key_jwt [OIDC-CORE]**.

damikael committed:
upd Section 4.2, 4.3

Section 4.2 of iGOV  the scopes **openid**, **offline_access** are used in SPID OpenID Connect proposal and they do not consider the other recommended scopes in the iGov profile, namely: **doc**.

Section 4.3 of iGov defines the policy regarding the userinfo object of claim request parameter. In CIE id, we define the policy both for the userinfo and ID Token object.

damikael committed:
upd sezione 4.2, sezione 4.3

 - Nella sezione 4.2 di iGOV gli scope **openid**, **offline_access** vengono usati in SPID OpenID Connect proposal e non considerano gli altri scope raccomandati nel profilo iGov, cioè: **doc**.

 - La sezione 4.3 di iGov definisce la politica relativa all'oggetto userinfo del claim request. In CIE id, definiamo la politica per entrambi gli oggetti userinfo e ID Token.

peppelinux commented on docs/en/seccons_bcps.rst:
```diff
@@ -36,7 +36,7 @@ Resolve endpoint
 
 This endpoint MUST release the Metadata, the Trust Marks and the previously processed Trust Chain, and MUST
-NOT trigger a procedure of Federation Entity Discovery for each request arrival.
+NOT trigger a procedure of Federation Entity Discovery for each request arrival, unless this endpoint is secured with a client authentication mechanism, such as private_key_jwt [OIDC-CORE].
```
questo è definito nelle considerazioni di sicurezza qui
https://openid.net/specs/openid-connect-federation-1_0.html#name-denial-of-service-attack-pr

eppoi l'endpoint può essere protetto in tanti modi, eventualmente dovremmo dire che se protetto con private_key_jwt, il metadata risolto DEVE riferirsi al subject contenuto nel private_key_jwt

altrimenti un soggetto X innescherebbe migliaia di discovery per metadata di altri soggetti, usando la sua private_key_jwt

peppelinux replied:
@fmarino-ipzs per te ok?

fmarino-ipzs replied:
Sono d'accordo con la possibilità di innescare la procedura di Discovery nel caso di autenticazione private_key_jwt ma inserendo il suggerimento di @peppelinux di risolvere solo per il sub contenuto nel private_key_jwt

peppelinux replied:
ok per me

AntonioFlorio replied:
Mi sembra ragionevole, chiedo anche @damikael se è d'accordo.

damikael replied:
Anche per me è ragionevole. In ogni caso, la modifica era per ora necessaria al fine di allineare il testo alla determina di AgID. A seguire potremmo anche ragionare insieme se lasciare o meno tale possibilità.

peppelinux commented on docs/it/log_management.rst:
```diff
@@ -20,6 +20,18 @@ Gli OP e gli RP DEVONO mantenere
     - L'eventuale **RevocationResponse** relativa alla *RevocationRequest*
+    
+.. admonition:: |spid-icon|
+    Per ogni messaggio, ove applicabili, potrebbero essere indicizzate ai fini di ricerca e consultazione le seguenti
```
potrebbero -> dovremmo usare linguaggio normativo

POSSONO

AntonioFlorio replied:
Ok per me

peppelinux replied:
`ove applicabile` non plurale

damikael replied:
Ok per me

peppelinux replied:
eccolo 63405d1f56e5a005abd2c757e9bf678309a37a83

peppelinux commented on docs/it/metadata_oidc_op.rst:
```diff
@@ -129,10 +129,10 @@ L'EC di un OP DEVE configurare un metadata di tipo **"openid_provider"** DEVE co
     - |spid-icon| |cieid-icon|
   * - **request_object_encryption_alg_values_supported**
-    - Vedi `OpenID.Discovery#OP_Metadata`_. Vedi key encryption :ref:`supported_algs`.
+    - Fino a diversa indicazione di AgID, non deve essere incluso.
```
questo testo non è neutrale per CIE

inoltre perché NON DEVE?
Se il RP supportasse la criptazione gli OP non devono supportarla?

AntonioFlorio replied:
Gli OP SPID al momento non supporteranno questa cifratura, quindi gli RP non possono chiederla.

peppelinux replied:
LLGG SPID OIDC supporta questo parametro

AntonioFlorio replied:
Vedi in https://github.com/italia/spid-cie-oidc-docs/pull/144#discussion_r1058153905 per request_object_encryption_enc_values_supported

peppelinux commented on docs/it/metadata_oidc_op.rst:
```diff
@@ -129,10 +129,10 @@ L'EC di un OP DEVE configurare un metadata di tipo **"openid_provider"** DEVE co
     - |spid-icon|
   * - **request_object_encryption_enc_values_supported**
-    - Vedi `OpenID.Discovery#OP_Metadata`_. Vedi Content Encryption :ref:`supported_algs`.
+    - Fino a diversa indicazione di AgID, non deve essere incluso.
```
come commento precedente

AntonioFlorio replied:
come risposta precedente

peppelinux replied:
LLGG SPID OIDC supporta questo parametro

AntonioFlorio replied:
Vedi in https://github.com/italia/spid-cie-oidc-docs/pull/144#discussion_r1058153905 per request_object_encryption_enc_values_supported

peppelinux commented on docs/it/metadata_oidc_rp.rst:
```diff
@@ -96,6 +96,9 @@ Il Metadata di tipo **"openid_relying_party"** DEVE contenere almeno i seguenti
+    - |spid-icon| |cieid-icon|
 
 .. note:: 
   Gli URI presenti nel parametro **redirect_uris** POSSONO anche seguire eventuali schemi custom (ad es. myapp://) al fine di supportare applicazioni mobili.
```
seguire -> usare/adottare

AntonioFlorio replied:
Si vuole evitare che il "response_types" possa essere utilizzato con valori diversi da "code".

Per me "usare"

damikael replied:
Ok per me "usare"

fmarino-ipzs replied:
Ritengo che quanto riportato nella nota sia misleading. Il supporto ad applicazioni mobili credo sia out of scope. Qui consideriamo solo client confidential e di tipo web. Il supporto mobile nativo va indirizzato correttamente e non tramite una nota messa qui.

peppelinux commented on docs/it/seccons_bcps.rst:
```diff
@@ -31,7 +31,7 @@ Allo stesso tempo la specifica OIDC Federation 1.0 non definisce un limite per i
 ++++++++++++++++
 
-Questo endpoint DEVE rilasciare i Metadata, i Trust Mark e la Trust Chain già precedentemente elaborata e NON DEVE innescare una procedura di Federation Entity Discovery ad ogni richiesta pervenuta.
+Questo endpoint DEVE rilasciare i Metadata, i Trust Mark e la Trust Chain già precedentemente elaborata e NON DEVE innescare una procedura di Federation Entity Discovery ad ogni richiesta pervenuta, a meno che questo endpoint non venga protetto con un meccanismo di autenticazione dei client, come ad esempio private_key_jwt [OIDC-CORE].
```
vedi commento precedente relativa alla versione inglese

AntonioFlorio replied:
Vedi risposta precedente.

peppelinux commented on docs/it/token_endpoint.rst:
```diff
@@ -223,7 +223,7 @@ Di seguito i claim che compongono l'Access Token.
 ++++++++
 
-L'ID Token è un JSON Web Token (JWT) che contiene informazioni sull'utente che ha eseguito l'autenticazione. I RP DEVONO eseguire la validazione dell'ID Token.
+L'ID Token è un JSON Web Token (JWT) firmato che contiene informazioni sull'utente che ha eseguito l'autenticazione. I RP DEVONO eseguire la validazione dell'ID Token.
```
eseguire la validazione -> devono validare il contenuto e la firma dell'ID Token

AntonioFlorio replied:
Per me, se un documento è firmato occorre sempre verificare contenuto e firma, altrimenti non ha senso firmare.

peppelinux replied:
effettivamente hai ragione, purtroppo posso firmare correttamente delle informazioni errate, come ad esempio un subject diverso

se un RP rilevasse anomalie di questo tipo questa verifica andrebbe oltre quello della firma digitale

si pensi ad un OP che ha il system time errato ed emette iat ed exp sfasati

fmarino-ipzs committed:
feat: Logout and revocation update as PR#131

fmarino-ipzs committed:
feat: added Refresh Token sec

peppelinux committed:
fix: minor editorials

fmarino-ipzs committed:
feat: scope and client_id param in AT PR#132

peppelinux committed:
fix: small editorials

peppelinux commented on docs/en/metadata_oidc_op.rst:
```diff
@@ -129,10 +129,10 @@ The EC of an OP MUST configure a metadata of type **"openid_provider"**, that MU
     - |spid-icon|
   * - **request_object_encryption_enc_values_supported**
-    - See `OpenID.Discovery#OP_Metadata`_. See Content Encryption :ref:`supported_algs`.
+    - Until otherwise indicated by AgID, this MUST NOT be included.
```
this is not CIE netrual

peppelinux replied:
@fmarino-ipzs vuoi dire qualcosa qui per cie?

fmarino-ipzs replied:
Eliminerei il riferimento esplicito alla AgID così da lasciare aperta la possibilità che a normarlo possa essere anche il Min Int per la CIE. Detto questo, non riesco a capire bene la ratio di questa frase. Inserite un parametro in una tabella che riporta i parametri del metadata OP, nella colonna "supported by" mettete SPID e poi dite che NON DEVE essere incluso??? Tanto vale non metterlo proprio. Quando AgID lo riterrà opportuno lo aggiungerà nella tabella e da quel momento gli OP dovranno prevederlo e supportarlo.

AntonioFlorio replied:
"request_object_encryption_enc_values_supported" è previsto nelle LLGG OIDC SPID, quindi è necessario chiarire se e come deve essere usato, abbiamo ritenuto che sia una modifica formale, altrimenti bisognava modificare le LLGG con i relativi tempi di attesa.

fmarino-ipzs replied:
@AntonioFlorio , non è possibile fare, come già fatto altre volte in passato, non riportarlo qui e dare questa indicazione in un avviso che mi pare di capire comunque uscirà? Io sarei per avere qui una visione di quello che è e non di quello che NON è e che forse sarà. Ovviamente è una mia personale riflessione e non è bloccante per noi visto che è qualcosa di supportato (non ancora) solo da SPID.

fmarino-ipzs replied:
In ogni caso modificherei in "It MUST NOT be included, unless otherwise specified"

peppelinux replied:
mi sembra di capire che è supportato da LLGG SPID OIDC, pertanto la dicitura "until otherwise" risulterebbe in contrasto con le LLGG SPID OIDC

decidiamo il testo da adottare qui @AntonioFlorio

AntonioFlorio replied:
Stiamo riportando in docsitalia quelle che abbiamo valutato come modifiche formali alle LLGG, per uniformare il più possibile SPID e CIE, sin da ora, senza dover attivare anche per queste cose l'iter ex art. 71 del CAD. 
Essendo, comunque, una modifica alle LLGG, è necessario esplicitarla e formalizzarla in maniera chiara, infatti la stessa indicazione è stata riportata anche nell'Avviso 41.
Per me va bene: "It MUST NOT be included, unless otherwise specified".

AntonioFlorio replied:
> @AntonioFlorio , non è possibile fare, come già fatto altre volte in passato, non riportarlo qui e dare questa indicazione in un avviso che mi pare di capire comunque uscirà? Io sarei per avere qui una visione di quello che è e non di quello che NON è e che forse sarà. Ovviamente è una mia personale riflessione e non è bloccante per noi visto che è qualcosa di supportato (non ancora) solo da SPID.

Se in questo documento non è necessario specificare per gli sviluppatori le differenze rispetto alle LLGG possiamo anche eliminare il "request_object_encryption_enc_values_supported".

peppelinux replied:
Risolto da questo

https://github.com/italia/spid-cie-oidc-docs/pull/144/commits/cf6f9f5b127d9514c2daa95eb2f4e934f351e694

peppelinux commented on docs/it/confronto_oidc_cie_e_oidc_igov.rst:
```diff
@@ -29,9 +29,11 @@ CIE OpenID Connect e SPID OpenID Connect sono basati su `iGov.OIDC`_ con le segu
  - Sezione 3.8. La registrazione dinamica di iGOV specifica che la registrazione dinamica del client è obbligatoria. Sia in CIE id che in SPID, la registrazione automatica OIDC del client è OBBLIGATORIA, mentre la registrazione dinamica OIDC del client NON DOVREBBE essere supportata.
 
- - Nella sezione 4.2 di iGOV gli scope **openid**, **offline_access**, **profile** e **email** vengono usati sia in SPID che in CIE id OpenID Connect proposal e non considerano gli altri scope raccomandatinel profilo iGov, cioè: **doc**.
+ - Nella sezione 4.2 di iGOV gli scope **openid**, **offline_access**, **profile** e **email** vengono usati in CIE id OpenID Connect proposal e non considerano gli altri scope raccomandati nel profilo iGov, cioè: **doc**.
```
"proposal" è un typo?

peppelinux replied:
fixed

peppelinux commented on docs/it/confronto_oidc_cie_e_oidc_igov.rst:
```diff
@@ -29,9 +29,11 @@ CIE OpenID Connect e SPID OpenID Connect sono basati su `iGov.OIDC`_ con le segu
+ - Nella sezione 4.2 di iGOV gli scope **openid**, **offline_access**, **profile** e **email** vengono usati in CIE id OpenID Connect proposal e non considerano gli altri scope raccomandati nel profilo iGov, cioè: **doc**.
 
- - La sezione 4.3 di iGov definisce la politica relativa all'oggetto userinfo del claim request. Sia in SPID che in CIE id, definiamo la politica per entrambi gli oggetti userinfo e ID Token.
+ - Nella sezione 4.2 di iGOV gli scope **openid**, **offline_access** vengono usati in SPID OpenID Connect proposal e non considerano gli altri scope raccomandati nel profilo iGov, cioè: **doc**.
```
"proposal" è un typo?

peppelinux committed:
fix: small editorials - reverting a removal

peppelinux commented on docs/en/revocation_endpoint.rst:
```diff
@@ -1,16 +1,31 @@
+  When the user logs out or when his/her session with the RP expires (according to the policies decided by the RP) the RP MUST revoke the Access Token in its possession.
+
+  .. note::
+    The Access Token revocation implies revoking all the Refresh Tokens linked to it.
```
qui non lo scriverei come una nota ma come un comportamento normato, DEVE revocare tutti i token ...

AntonioFlorio replied:
Per me ok.

damikael replied:
Ok per me

peppelinux replied:
cari, nota implementativa, DEVE revocare SOLO SE l'Access Token non fosse scaduto.
Se scaduto non serve la revoca, right?

Qui 0c4d029d6535812a0d413407404d8dfb4c880e26

AntonioFlorio replied:
Se la domanda è riferita alla revoca di un Access Token scaduto, la revoca mi sembra inutile. 
Se, invece, la domanda è riferita alla revoca del Refresh Token alla scadenza dell'Access Token, mi sembra che viene meno il meccanismo di refresh.

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -205,11 +205,17 @@ UserInfo endpoint to get user attributes.
      - |spid-icon| |cieid-icon|
    * - **sub** 
-     - See `OpenID.Core#SubjectIDTypes`_. It MUST be *pairwise*. 
+     - See `OpenID.Core#SubjectIDTypes`_. It MUST be *pairwise* type. 
```
`It MUST be *pairwise* type` è un typo

it must be pairwised oppure it must be of pairwise type

dobbiamo dire che il valore deve essere di tipo pairwised (con ref a OIDC core)

AntonioFlorio replied:
Per me ok entrambe.

damikael replied:
Preferisco "of pairwise type"

peppelinux replied:
fatto qui 03640c32f64041d5d9388a5367cabf1ac0a47755

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -205,11 +205,17 @@ UserInfo endpoint to get user attributes.
+     - See `OpenID.Core#SubjectIDTypes`_. It MUST be *pairwise* type. 
+     - |spid-icon| |cieid-icon|
+   * - **client_id** 
+     - It MUST contain an HTTPS URL that uniquely identifies the RP. 
```
typo

`a HTTPs`

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -205,11 +205,17 @@ UserInfo endpoint to get user attributes.
+     - The OP SHOULD add the *scope* parameter as defined in :rfc:`9068` Section 2.2.3. It MUST match the value in the authentication request.
+     - |spid-icon| |cieid-icon|
    * - **iat** 
      - UNIX Timestamp with the time of JWT generation, coded as NumericDate as indicated in :rfc:`7519`. 
```
`generation` -> `issuance`

fmarino-ipzs replied:
done

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -226,7 +232,7 @@ UserInfo endpoint to get user attributes.
 ++++++++
 
-The ID Token is a JSON Web Token (JWT) that contains information on the user that has executed the authentication. The RPs MUST validate the ID Token.
+The ID Token is a signed JSON Web Token (JWT) that contains information on the user that has executed the authentication. The RPs MUST validate the ID Token.
```
`on the user` -> `about the user`

peppelinux commented on docs/it/log_management.rst:
```diff
@@ -20,6 +20,18 @@ Gli OP e gli RP DEVONO mantenere
+        - iat
+        - exp
 
 2. Un registro di federazione contenente, per ogni *AuthenticationRequest*, la **Trust Chain** relativa all'entità con la quale si stanno scambiando i messaggi. Nel caso di registro mantenuto da un OP che riceve una richiesta di autenticazione da un RP, DEVE contenere la seguente lista ordinata di oggetti:
```
DEVE è riferito a quale soggetto, il registro?

peppelinux commented on docs/it/metadata_oidc_op.rst:
```diff
@@ -129,10 +129,10 @@ L'EC di un OP DEVE configurare un metadata di tipo **"openid_provider"** DEVE co
     - |spid-icon|
   * - **request_object_encryption_enc_values_supported**
-    - Vedi `OpenID.Discovery#OP_Metadata`_. Vedi Content Encryption :ref:`supported_algs`.
+    - Fino a diversa indicazione di AgID, non deve essere incluso.
```
`di AgID` -> non era `della AgID`?

AntonioFlorio replied:
Se é sempre usato della AgID possiamo anche continuare.

peppelinux commented on docs/it/revocation_endpoint.rst:
```diff
@@ -1,16 +1,32 @@
+  .. note::
+    La revoca di un Access Token comporta la revoca di tutti i Refresh Token a questo collegati.
+
+  L'OP DOVRÀ revocare il token specificato nella richiesta e DOVRÀ terminare la sessione di Single Sign-On se ancora attiva. Eventuali altri token attivi per l’utente dovranno invece essere mantenuti validi.
```
definire Single Sign-On nei termini definiti, sappiamo che esistono diverse assunzioni del significato di questo termine

AntonioFlorio replied:
Ho proposto l'inserimento della seguente definizione e indicazione nell'Avviso SPID:

Nelle LLGG OIDC SPID con il termine “sessioni di SSO” si intende: la sessione di autenticazione associata ad un determinato utente titolare di identità digitale, mantenuta da un Gestore dell’identità digitale (OP) nei confronti di un singolo Fornitore di servizi (RP).

Fino all’emanazione di apposite regole tecniche non è prevista l’instaurazione di una sessione di autenticazione associata ad un determinato utente titolare di identità digitale, mantenute da un Gestore dell’identità digitale (OP) nei confronti di diversi Fornitori di servizi (RP).

peppelinux replied:
Propongo di averlo nei Defined terms

fmarino-ipzs committed:
fix: typo

fmarino-ipzs committed:
fix: removed SPID reference in Metadata Retrieval sec

fmarino-ipzs committed:
fix: typo in metadata policy

fmarino-ipzs committed:
feat: resolve endpoint sub MUST match sub in private_key_jwt

peppelinux committed:
fix: defined terms

peppelinux committed:
fix: Acronyms and editorials

peppelinux committed:
fix: editorials on introductionary federation text

peppelinux committed:
fix: editorials on EC

fmarino-ipzs committed:
fix: bug the acronyms table

fmarino-ipzs committed:
Merge branch 'spid-docs-conformance' of https://github.com/italia/spid-cie-oidc-docs into spid-docs-conformance

peppelinux committed:
fix: editorials on ES

peppelinux committed:
fix: editorials aggregatori

fmarino-ipzs committed:
fix: token endpoint introduction

fmarino-ipzs committed:
Merge branch 'spid-docs-conformance' of https://github.com/italia/spid-cie-oidc-docs into spid-docs-conformance

peppelinux committed:
fix: editorials trust neg

fmarino-ipzs committed:
fix: editorial in log management

fmarino-ipzs committed:
Merge branch 'spid-docs-conformance' of https://github.com/italia/spid-cie-oidc-docs into spid-docs-conformance

peppelinux committed:
fix: editorials fed endpoints

peppelinux committed:
fix: editorials op metadata jwks

peppelinux committed:
fix: metadata rp auth code

peppelinux commented on docs/en/metadata_oidc_op.rst:
```diff
@@ -168,7 +168,7 @@ The EC of an OP MUST configure a metadata of type **"openid_provider"**, that MU
 .. warning::
   The OP Metadata of type **"openid_provider"** exposes the claim **jwks** as regulated by OID-FED instead of
-  the claim **jwks_uri** as required at `OpenID.Discovery#OP_Metadata`_.
+  the claim **jwks_uri** as required at `OpenID.Discovery#OP_Metadata`_. However, for interoperability purpose, it MAY expose also the parameter **jwks_uri**.
```
noi diciamo CHE DEVE usare jwks ma considerando che una entità può partecipare ad altre federazioni non dobbiamo impedire un modello di interop

possiamo escludere jwks_uri, se necessario, con le metadata policy

AntonioFlorio replied:
Stiamo normando il metadata dell'OP in SPID e CIE non in altre federazioni. Inoltre, in quali casi un OP può operare anche in altre federazioni?

peppelinux committed:
fix: metadata rp custm schemas typo

peppelinux commented on docs/en/metadata_oidc_rp.rst:
```diff
@@ -95,6 +95,9 @@ The RP Metadata of type **"openid_relying_party"** MUST contain at least the fol
+    - |spid-icon| |cieid-icon|
 
 .. note:: 
   The URIs contained in the claim **redirect_uris** MAY also follow possible custom schemas (e.g. myapp://) 
```
sparita la review, cmq eccolo 8755f8b35d9e5816e9d7eba1e856053e1ffbd63d

peppelinux committed:
fix: revocation if not expired yet

peppelinux commented on docs/en/revocation_endpoint.rst:
```diff
@@ -1,16 +1,31 @@
+  .. note::
+    The Access Token revocation implies revoking all the Refresh Tokens linked to it, if not expired yet.
+
+  The OP will have to revoke the token specified in the request and will have to end the Single Sign-On session, if it is still active. Any other active tokens for the user must be kept valid.
```
non capisco a cosa si riferisce "Any other active tokens for the user must be kept valid."

se abbiamo detto che la revoca di un AT comporta la revoca di tutti i refresh collegati, quali altri token dovrebbero risultare attivi?

AntonioFlorio replied:
Mi sembra che qui il discorso fosse relativo alla revoca del Refresh Token, che non deve incidere su Access Token e ID token,  alla revoca dell'ID Token che non deve incidere sull'Access Token e viceversa.

peppelinux replied:
lasciamo as-is, in previsione dei futuri lavori su SSO ed estensione del refresh lato AgID.

peppelinux commented on docs/en/revocation_endpoint.rst:
```diff
@@ -1,16 +1,31 @@
+
+  The Access Token revocation MUST NOT imply revoking all the Refresh Tokens linked to it.
+
+  If the token passed to the request is a Refresh Token, the OP MUST revoke the respective Access Token as well, if it is not expired yet.
```
va bene "if it is not expired yet"? @fmarino-ipzs

fmarino-ipzs replied:
Sì concordo. Lo davo per scontato visto che non ha senso revocare un token scaduto..

peppelinux committed:
fix: pairwise typo

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -226,7 +234,7 @@ UserInfo endpoint to get user attributes.
-The ID Token is a JSON Web Token (JWT) that contains information on the user that has executed the authentication. The RPs MUST validate the ID Token.
+The ID Token is a signed JSON Web Token (JWT) that contains information about the user that has executed the authentication. The RPs MUST validate the ID Token.
 
 If the RP exposes in its metadata the parameter **id_token_encrypted_response_alg** the OP MUST encrypt the ID Token.
```
questo collide con "until otherwise" modificato dai colleghi agid

@damikael @AntonioFlorio ^

AntonioFlorio replied:
Questo è riferito a id_token_encrypted_response_alg?
Per il momento, in SPID il RP non può chiedere la cifratura dell'ID Token, perché non è supportata dagli OP.
Avevamo proposto di mantenere, per il momento, l'indicazione dell'id_token_encrypted_response_alg solo per CIE, per poter inserire la modifica nelle prossime modifiche alle LLGG OIDC.

peppelinux replied:
Risolto qui 405291beed015c6148fea39e8daeefbaeb61c366

peppelinux commented on docs/en/token_endpoint.rst:
```diff
@@ -302,6 +310,67 @@ The claims available in the *ID Token* are given below.
+
+The *Refresh Token* MUST NOT be valid (difference between *iat* and *exp*) for more than 30 days. 
+
+If at the expiration of the validity period the RP makes a request to the OP, the OP MUST return an error as the result of the request. 
```
typo "if at the"

agcolella replied:
Upon expiration of the validity ....

peppelinux replied:
done ee6313c3a4eb8491488513df0e61f2d9a86ef669

peppelinux committed:
fix: normative language typo

peppelinux committed:
fix: typo ove applicabili

peppelinux committed:
fix: request object enc algs rimosso / commentato fino a prossimo avviso SPID

peppelinux committed:
fix: jwks_uri removed as not of interest

peppelinux committed:
fix: revocation normative lang

peppelinux committed:
fix: comments on individual sessions [WiP]

peppelinux committed:
fix: typo "call me"

peppelinux committed:
fix: id_token_encrypted_response_alg disambiguazione spid/cie

peppelinux committed:
fix: jwt issuance, not generation

peppelinux committed:
fix: "for more" typo

peppelinux committed:
fix: "if at the" typo

peppelinux committed:
fix: "proposal" typo

peppelinux committed:
chore: log management, trust chain explanatory text

peppelinux commented:
Si richiede un ulteriore PR contenente:

1. termini definiti di sessione individuale e sessione di autenticazione
2. box disambiguazione spid/cie su refresh
